cli/cmd/encore/gen.go

```diff
@@ -37,15 +44,29 @@ Use '--env=local' to generate it based on your local development version of the
 Supported language codes are:
   typescript: A TypeScript-client using the in-browser Fetch API
   go: A Go client using net/http"
+
+Support presets are:
+	nextjs: Generates a TypScript client for use within a Next.js application
 `,
 		Args: cobra.ExactArgs(1),
 		Run: func(cmd *cobra.Command, args []string) {
-			if output == "" && lang == "" {
-				fatal("specify at least one of --output or --lang.")
+			if output == "" && lang == "" && preset == "" {
+				fatal("specify at least one of --output, --lang or --preset.")
```

Comment:
I am concerned that this makes for quite the analysis paralysis when trying to decide how to use this command. Could we combine `lang` and `preset` somehow to just a single option? We could have `--lang=ts` choose the default preset and `--lang=ts.nextjs` (or `--lang=ts/nextjs`) choose the nextjs preset, or whatever.